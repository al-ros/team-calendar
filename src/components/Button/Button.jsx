import PropTypes from 'prop-types'
import cn from 'classnames'
import './Button.scss'

const Button = ({onClick, label, className, disabled}) =>
    <button disabled={disabled} className={cn('button', className, {
        'button--disabled': disabled 
    })} onClick={onClick}>{label}</button>

Button.propTypes = {
    onClick: PropTypes.func,
    label: PropTypes.string,
    className: PropTypes.string,
    disabled: PropTypes.bool
}

export default Button
