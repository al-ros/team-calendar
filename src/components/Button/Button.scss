.button {
    display: inline-block;
    color: rgb(77, 77, 77);
    text-align: center;
    white-space: nowrap;
    user-select: none;
    border:  2px solid rgb(77, 77, 77);
    font-weight: 600;
    font-size: 1rem;
    border-radius: 0.25rem;
    min-width: 10rem;
    height: 2rem;
    background-color:#eee;
    cursor: pointer;
    
    &:active {
        background-color: #ddd;
    }

    &--disabled {
        pointer-events: none;
        opacity: .5;
    }
}
