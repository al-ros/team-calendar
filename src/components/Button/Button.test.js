import {render, screen} from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Button from './Button'

describe('<Button />', () => {
  it('should render Button component properly', () => {

    const { asFragment } = render(<Button />);
    const button = screen.getByRole('button');

    expect(button).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot(); // snapshot
  });

  it('should render Button with label', () => {
    render(<Button label="The magic button" />);
    const button = screen.getByRole('button', { name: /magic/i });
    expect(button).toBeInTheDocument();

    const button2 = screen.queryByRole('button', { name: /asdf/i });
    expect(button2).not.toBeInTheDocument();
  });

  it('should call onClick', () => {
    const handleClick = jest.fn(); // mock function

    render(<Button onClick={handleClick} label="The magic button" />);
    const button = screen.getByRole('button', { name: /magic/i });
    
    userEvent.click(button);

    expect(handleClick).toHaveBeenCalled();
  });

  // it('should be disabled', () => {
  //   expect(true).toBe(true);
  // })

  // it('should be block', () => {
  //   expect(true).toBe(true);
  // })

  // it('should has classname', () => {
  //   expect(true).toBe(true);
  // })
});
