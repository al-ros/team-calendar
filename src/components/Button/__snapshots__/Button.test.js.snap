// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render Button component properly 1`] = `
<DocumentFragment>
  <button
    class="button"
  />
</DocumentFragment>
`;
